Migrate to Apply API and refactor based on applications rather than applicants

mkitt opened the pull request:
🚧

mkitt committed:
Add staging endpoint and convert cards

This starts the work for migrating to a application based instead of
applicant based.

mkitt committed:
Remove unused mutation placeholder files

mkitt committed:
Convert the birthdate property

mkitt committed:
Updated failing and missing tests

srt32 commented on flow-typed/schema.gql:
```diff
@@ -10,7 +10,7 @@ type Applicant @model {
   action: Action @relation(name: "ApplicantOnAction")
   applications: [Application!]! @relation(name: "ApplicationOnApplicant")
-  birthdate: String
+  birthDate: String
```
I'm not sure what the API is returning but we recently switched to collecting only a birth year if that makes things easier.

mkitt replied:
We just use it for display so if there is a birth year available would be pretty easy to swap that out once it's available in the query.

mkitt committed:
Add more thorough middleware and error handling

mkitt committed:
Convert as much as we can in the other views

Continues moving to Applications instead of Applicant. Lots still
missing in the API so stubbing in placeholder data as well.

mkitt committed:
Add proper logging

mkitt committed:
Bumps dependencies

The jest-glamor package update caused the output in the snapshots to be
invalidated :(

mkitt commented:
Merging this in to make sure the deployments are setup correctly to staging. Production will still have the old endpoint for reference until we tag a release though.
